packages/grid/x-grid-modules/src/hooks/utils/useOptionsProp.ts

```diff
@@ -0,0 +1,107 @@
+import {
+  GridComponentProps,
+  GridOptionsProp,
+} from '@material-ui/x-grid-modules/gridComponentProps';
+import { DEFAULT_GRID_OPTIONS, GridOptions } from '@material-ui/x-grid-modules/models';
+import * as React from 'react';
+import { mergeOptions } from '@material-ui/x-grid-modules/utils';
+
+export function useOptionsProp(props: GridComponentProps): [GridOptions, Function] {
```

Comment:
I don't understand the value of this hook. Is there a way we could break it down? I don't see how this can scale to more props.

Reply:
This hook aims to restructure the props and also avoids having too much impact on the grid functionality. The grid is built on many effect and options dependencies, so by restructuring at this level, I avoid a high impact refactoring right before the release of our MVP product. This is a less risky approach. 
Later on we can refactor, by restructuring in smaller object such as events pagination... Or just manipulate individual props.

Reply:
Oh I see, so it's meant to avoid having to refactor the internal. Should we add a note about it somewhere, for the post MVP? It seems that we could make the codebase easier to follow without this indirection :)

Reply:
Ok we can add a note somewhere. But where? Github? Is it time to discuss our process?

Reply:
I think that it depends on how much we want to get the changes done.

If it's something we only don't want to miss in the next refactorization, I think that a `// TODO` is enough. If it's something we want to have it down, a new issue on GitHub can do the trick.

Reply:
Ok I'll add a todo in the code.